### ISSUE TITLE
New lint: `derived_partial_eq_with_manual_eq`

### ISSUE THREAD

**Issue by tyilo** [rust-lang/rust-clippy#17685]
### What it does

Disallows a type from having a derived `PartialEq` with a manual `Eq` implementation.

### Advantage

- Prevents accidental wrong `Eq` implementations for types containing floats (for instance: https://github.com/mongodb/bson-rust/issues/683)

### Example

```rust
#[derive(PartialEq)]
struct Foo {
    value: u32,
}

impl Eq for Foo {}

#[derive(PartialEq)]
struct Bar {
    value: f32,
}

impl Eq for Bar {}
```

Either both `PartialEq` and `Eq` should be manually implemented or both should be derived:
```rust
#[derive(PartialEq, Eq)]
struct Foo {
    value: u32,
}

struct Bar {
    value: f32,
}

impl PartialEq for Bar {
    fn eq(&self, other: &Self) -> bool {
        return self.value.to_bits() == other.value.to_bits()
    }
}

impl Eq for Bar {}
```

Note that the following doesn't compile:
```rust
#[derive(PartialEq, Eq)]
struct Bar {
    value: f32,
}
```
